source/Ship.cpp

```diff
@@ -3125,21 +3152,29 @@ bool Ship::HasBays() const
 // one of your escorts plans to use that bay.
 int Ship::BaysFree(const string &category) const
 {
-	int count = 0;
-	for(const Bay &bay : bays)
-		count += (bay.category == category) && !bay.ship;
-	return count;
+	auto countFunc = [&category](const Bay &bay) -> bool {
+		return bay.CanContain(category) && !bay.ship;
+	};
+	return std::count_if(bays.begin(), bays.end(), countFunc);
```

Comment:
```suggestion
	return std::count_if(bays.begin(), bays.end(), [&category](const Bay &bay) -> bool
	{
		return bay.CanContain(category) && !bay.ship;
	});
```
I would normally do it like this, but do whatever you want

Reply:
I'll leave it as is since it's inconsequential.